src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1288,145 +1271,175 @@ MonodroidRuntime::set_environment_variable_for_directory (const char *name, jstr
 }
 
 inline void
-MonodroidRuntime::create_xdg_directory (jstring_wrapper& home, const char *relativePath, const char *environmentVariableName)
+MonodroidRuntime::create_xdg_directory (jstring_wrapper& home, size_t home_len, const char *relativePath, size_t relative_path_len, const char *environmentVariableName)
 {
-	simple_pointer_guard<char> dir = utils.path_combine (home.get_cstr (), relativePath);
-	log_info (LOG_DEFAULT, "Creating XDG directory: %s", static_cast<char*>(dir));
-	int rv = utils.create_directory (dir, DEFAULT_DIRECTORY_MODE);
+	static_local_string<SENSIBLE_PATH_MAX> dir (home_len + relative_path_len);
+	utils.path_combine (dir, home.get_cstr (), home_len, relativePath, relative_path_len);
+	log_info (LOG_DEFAULT, "Creating XDG directory: %s", dir.get ());
+	int rv = utils.create_directory (dir.get (), DEFAULT_DIRECTORY_MODE);
 	if (rv < 0 && errno != EEXIST)
-		log_warn (LOG_DEFAULT, "Failed to create XDG directory %s. %s", static_cast<char*>(dir), strerror (errno));
+		log_warn (LOG_DEFAULT, "Failed to create XDG directory %s. %s", dir.get (), strerror (errno));
 	if (environmentVariableName)
-		setenv (environmentVariableName, dir, 1);
+		setenv (environmentVariableName, dir.get (), 1);
 }
 
 inline void
 MonodroidRuntime::create_xdg_directories_and_environment (jstring_wrapper &homeDir)
 {
-	create_xdg_directory (homeDir, ".local/share", "XDG_DATA_HOME");
-	create_xdg_directory (homeDir, ".config", "XDG_CONFIG_HOME");
+	size_t home_len = strlen (homeDir.get_cstr ());
+
+	constexpr char XDG_DATA_HOME[] = "XDG_DATA_HOME";
+	constexpr char HOME_PATH[] = ".local/share";
+	constexpr size_t HOME_PATH_LEN = sizeof(HOME_PATH) - 1;
+	create_xdg_directory (homeDir, home_len, HOME_PATH, HOME_PATH_LEN, XDG_DATA_HOME);
+
+	constexpr char XDG_CONFIG_HOME[] = "XDG_CONFIG_HOME";
+	constexpr char CONFIG_PATH[] = ".config";
+	constexpr size_t CONFIG_PATH_LEN = sizeof(CONFIG_PATH) - 1;
+	create_xdg_directory (homeDir, home_len, CONFIG_PATH, CONFIG_PATH_LEN, XDG_CONFIG_HOME);
 }
 
 #if DEBUG
 void
 MonodroidRuntime::set_debug_env_vars (void)
 {
-	char *value;
-
-	if (androidSystem.monodroid_get_system_property (Debug::DEBUG_MONO_ENV_PROPERTY, &value) == 0)
+	dynamic_local_string<PROPERTY_VALUE_BUFFER_LEN> value;
+	if (androidSystem.monodroid_get_system_property (Debug::DEBUG_MONO_ENV_PROPERTY, value) == 0)
 		return;
 
-	char **args = utils.monodroid_strsplit (value, "|", 0);
-	delete[] value;
+	auto log_envvar = [](const char *name, const char *v) {
+		log_info (LOG_DEFAULT, "Env variable '%s' set to '%s'.", name, v);
```

Comment:
I don't think we previously logged this info; any reason to begin doing so?

Reply:
It should use `log_debug` and the reason is to double-check that parsing is correct, if need be.